### PR TITLE
feat: profile page

### LINKED ISSUE #77: [Feat] : Profile Page
## Describe

There is no profile page or modal which gives user info.

Please assign me this issue under GSSOC'24

### PR DESCRIPTION
**Pull request by Sid-80**
## Describe
Profile page

## Issue
resolves #77 

image

image

image

### CODE REVIEW

**Review by pooranjoyb — COMMENTED** [review]
Great job, @Sid-80! 🎉 Thank you for submitting your pull request. Your contribution is valuable and we appreciate your efforts to improve our project.<p>We will promptly review your changes and offer feedback. Keep up the excellent work! Kindly remember to check our [contributing guidelines](https://github.com/pooranjoyb/popShop/blob/dev/CONTRIBUTING.md)</p>

**Comment by Sid-80** [comment]
@pooranjoyb Please review it!

File: src/pages/Profile/Profile.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    if (!isAuthenticated) {
+        navigate('/home');
+    }
+
+    return (
+        <div className="flex relative flex-col gap-5 items-center justify-center px-10 lg:px-[300px]">
+            <img className="absolute opacity-30 -z-20 w-full h-[220px] sm:h-[350px] top-[-98px]" src="/images/backbest.jpeg" alt="" />
+            <div className="divider divider-neutral text-lg font-semibold sm:text-3xl divider-start">
+                <FaUser width={20} />
```
Comment (pooranjoyb) [inline src/pages/Profile/Profile.tsx:21]: remove this icon

File: src/pages/Profile/Profile.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                Profile
```
Comment (pooranjoyb) [inline src/pages/Profile/Profile.tsx:22]: Profile text should use `<Head>`

File: src/pages/Profile/Profile.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            </div>
+            <div className="avatar">
+                <div className="w-24 sm:w-80 rounded-full">
+                    <img src="/images/winter2.jpg" alt="" />
+                </div>
+            </div>
+            <div className="w-full text-center">
+                <Head nextLine={true} h1={username!} h2="First Name Last Name" />
```
Comment (pooranjoyb) [inline src/pages/Profile/Profile.tsx:30]: remove this h2 and add the first anme and last name in the details below. 
Only username should be there with `Head`

File: src/components/Head.tsx [hunk 4]
```diff
@@ -1,13 +1,15 @@
...
+  nextLine?:boolean;
 }
 
-export default function Head({ h1, h2 }: HeadProps) {
+export default function Head({ h1, h2,nextLine }: HeadProps) {
   return (
     <div className="flex text-mynavy justify-center items-center">
       <div className="text-4xl sm:text-6xl font-extrabold">
         {h1}
+        {(nextLine) ? <br /> : ""}
```
Comment (pooranjoyb) [inline src/components/Head.tsx:12]: no need next line, use multiple `<Head>` when needed.

File: src/pages/Profile/Profile.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+                        </div>
+                        <div className="flex sm:flex-row flex-col items-start sm:items-center  text-lg sm:text-xl justify-start">
+                            <div className="label">
+                                <span className="label-text text-xl">Account Creation Date : </span>
+                            </div>
+                            <h1 className=" font-bold text-mynavy">1/1/24</h1>
+                        </div>
+                    </div>
+
+                    <div className="hidden sm:block border border-l-[1px]" />
```
Comment (pooranjoyb) [inline src/pages/Profile/Profile.tsx:56]: this divider should be a but lighter, use `gray-400`

File: src/pages/Profile/Profile.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    const navigate = useNavigate();
+
+    if (!isAuthenticated) {
+        navigate('/home');
+    }
+
+    return (
+        <div className="flex relative flex-col gap-5 items-center justify-center px-10 lg:px-[300px]">
+            <img className="absolute opacity-30 -z-20 w-full h-[220px] sm:h-[350px] top-[-98px]" src="/images/backbest.jpeg" alt="" />
+            <div className="divider divider-neutral text-lg font-semibold sm:text-3xl divider-start">
```
Comment (pooranjoyb) [inline src/pages/Profile/Profile.tsx:20]: romove divider

**Review by pooranjoyb — CHANGES_REQUESTED** [review]

File: src/pages/Profile/Profile.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+    const navigate = useNavigate();
+
+    if (!isAuthenticated) {
+        navigate('/home');
+    }
+
+    return (
+        <div className="flex relative flex-col gap-5 items-center justify-center px-10 lg:px-[300px]">
+            <img className="absolute opacity-30 -z-20 w-full h-[220px] sm:h-[350px] top-[-98px]" src="/images/backbest.jpeg" alt="" />
```
Comment (pooranjoyb) [inline src/pages/Profile/Profile.tsx:19]: remove this image, possibly add a `Linear-Gradient` with the theme of the PopShop. 
Check `tailwind.config.js`

**Review by pooranjoyb — CHANGES_REQUESTED** [review]

**Review by pooranjoyb — APPROVED** [review]

**Comment by pooranjoyb** [comment]
Nice work with the page @Sid-80 🚀

**Pull request merged**